src/appleseed/renderer/kernel/shading/shadingpoint.h

```diff
@@ -188,6 +188,9 @@ class ShadingPoint
     // Return the material at the intersection point, or 0 if there is none.
     const Material* get_material() const;
 
+    // Return the back material at the intersection point, or 0 if there is none.
+    const Material* get_opposite_material() const;
```

Comment:
I think `get_front_material()` / `get_back_material()` would be nicer and clearer.